lib/handlers/delete.js

```diff
@@ -16,9 +16,7 @@ function handler(req, res) {
     ldp.delete(filename, function(err) {
         if (err) {
             debug("DELETE -- error: " + err);
-            return res
-                .status(err.status)
-                .send(err.message);
+            return next(err);
```

Comment:
This comment is valid for all of your other changes,
Are you sure this is necessary?

It may be I am super wrong, but you could just intercept every time a red.send(404) or similar happens like in my example